presto-main/src/test/java/com/facebook/presto/cost/TestCostCalculator.java

```diff
@@ -0,0 +1,397 @@
+    }
+
+    private void assertCostHasUnknownComponentsForUnknownStats(PlanNode node)
+    {
+        new CostAssertionBuilder(calculateCumulativeCost(
+                costCalculatorUsingExchanges,
+                node,
+                planNode -> UNKNOWN_COST,
+                planNode -> UNKNOWN_STATS))
+                .hasUnknownComponents();
+        new CostAssertionBuilder(calculateCumulativeCost(
+                costCalculatorWithEstimatedExchanges,
+                node,
+                planNode -> UNKNOWN_COST,
+                planNode -> UNKNOWN_STATS))
+                .hasUnknownComponents();
+    }
+
+    private PlanNodeCostEstimate calculateCumulativeCost(
+            CostCalculator costCalculator,
+            PlanNode node,
+            Function<PlanNode, PlanNodeCostEstimate> costs,
+            Function<PlanNode, PlanNodeStatsEstimate> stats)
+    {
+        PlanNodeCostEstimate localCost = costCalculator.calculateCost(
+                node,
+                planNode -> requireNonNull(stats.apply(planNode), "no stats for node"),
+                noLookup(),
+                session,
+                ImmutableMap.of());
+
+        PlanNodeCostEstimate sourcesCost = node.getSources().stream()
+                .map(source -> requireNonNull(costs.apply(source), "no cost for source"))
+                .reduce(ZERO_COST, PlanNodeCostEstimate::add);
+        return sourcesCost.add(localCost);
+    }
+
+    private static class CostAssertionBuilder
+    {
+        private final PlanNodeCostEstimate actual;
+
+        CostAssertionBuilder(PlanNodeCostEstimate actual)
+        {
+            this.actual = requireNonNull(actual, "actual is null");
+        }
+
+        CostAssertionBuilder cpu(double value)
+        {
+            assertEquals(actual.getCpuCost(), value, 0.000001);
+            return this;
+        }
+
+        CostAssertionBuilder memory(double value)
+        {
+            assertEquals(actual.getMemoryCost(), value, 0.000001);
+            return this;
+        }
+
+        CostAssertionBuilder network(double value)
+        {
+            assertEquals(actual.getNetworkCost(), value, 0.000001);
+            return this;
+        }
+
+        CostAssertionBuilder hasUnknownComponents()
+        {
+            assertTrue(actual.hasUnknownComponents());
+            return this;
+        }
+    }
+
+    private static PlanNodeStatsEstimate statsEstimate(int outputSizeInBytes)
+    {
+        double rowCount = Math.max(outputSizeInBytes / 8, 1);
+
+        return PlanNodeStatsEstimate.builder()
+                .setOutputRowCount(rowCount)
+                .addSymbolStatistics(
+                        new Symbol("s"),
+                        SymbolStatsEstimate.builder()
+                                .setAverageRowSize(outputSizeInBytes / rowCount)
+                                .build())
+                .build();
+    }
+
+    private TableScanNode tableScan(String id, String... symbols)
```

Comment:
Why not to use `PlanBuilder` here?

Reply:
can i ignore this comment?

Reply:
wait, I want to know the answer too.  It doesn't make sense to have different tests building nodes in different ways unless there is a reason.

Reply:
There is currently zero-overlap between what is needed here and what is provided by `PlanBuilder`. I am open to suggestion, but it seems like using `PlanBuilder` here would make these tests simpler.